mmv1/third_party/terraform/main.go.erb

```diff
@@ -2,11 +2,57 @@
 package main
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+	"context"
+	"flag"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server"
+	"github.com/hashicorp/terraform-plugin-mux/tf5muxserver"
 	"github.com/hashicorp/terraform-provider-google<%= "-" + version unless version == 'ga'  -%>/google<%= "-" + version unless version == 'ga'  -%>"
 )
 
+var (
+	// these will be set by the goreleaser configuration
+	// to appropriate values for the compiled binary
+	version string = "dev"
+
+	// goreleaser can also pass the specific commit if you want
+	// commit  string = ""
+)
```

Comment:
Does anything need to be changed with [version/version.go](https://github.com/hashicorp/terraform-provider-google/blob/main/version/version.go) and the release workflows to work with the muxed provider?

Reply:
Just surfacing info : Here's where `version.ProviderVersion` is used in the code:

https://github.com/GoogleCloudPlatform/magic-modules/blob/3456e2c8a2882be6ccb585401fba57a91ea2b3ac/mmv1/third_party/terraform/utils/provider.go.erb#L536-L540

And here's where the goreleaser process overrides the value of that variable in the release process:

https://github.com/hashicorp/terraform-provider-google/blob/5bc7515c1a5b8e2248222be77386a894414717a8/.goreleaser.yml#L31-L32

---

Could you use `version.ProviderVersion` instead of this new version string? Or is this the new way things are done for framework plugins?

Reply:
Copying over details from a Slack conversation:

The new variables `version` and `commit` need to be reflected in the goreleaser file to let them be over-written. [This is the line in the framework scaffold repo that allows their values to be overwritten vid ldsflags argument.](https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/b6f0e7fef1edffef4eb01a1b5e31122edf7d7bf1/.goreleaser.yml#L17)

In the old SDK code we get info about what version is being released via a version package and a var called `ProviderVersion`. [In goreleaser files for SDK providers the ldflags argument is different, and overwrites that var instead.](https://github.com/hashicorp/terraform-provider-google/blob/5bc7515c1a5b8e2248222be77386a894414717a8/.goreleaser.yml#L32)

If our muxed provider is going to have [this code unchanged](https://github.com/GoogleCloudPlatform/magic-modules/blob/3456e2c8a2882be6ccb585401fba57a91ea2b3ac/mmv1/third_party/terraform/utils/provider.go.erb#L536-L540) and still using `version.ProviderVersion` and also use the new `version` var in the google package, then we'd need to create a new combined version of the ldflags argument like this (**though I'd want to check it manually first**):

```
- '-s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X github.com/hashicorp/terraform-provider-google/version.ProviderVersion={{.Version}}'
```

Reply:
Addressed in 1ce4eb6 - I've manually confirmed that it creates the framework user agent OK